resources/qml/dialogs/LeaveRoomDialog.qml

```diff
@@ -18,5 +19,18 @@ P.MessageDialog {
     text: qsTr("Are you sure you want to leave?")
     modality: Qt.ApplicationModal
     buttons: P.MessageDialog.Ok | P.MessageDialog.Cancel
-    onAccepted: Rooms.leave(roomId, reason)
+    onAccepted: {
+
+        if (CallManager.haveCallInvite) {
```

Comment:
I think you should handle that on the c++ side, since the leave can be triggered in multiple ways.